src/AddTodoForm.jsx

```diff
@@ -1,22 +1,30 @@
-const AddTodoForm = (props) => {
+import { useState } from 'react';
+
+const AddTodoForm = ({onAddTodo}) => {
+    let [todoTitle, setTodoTitle] = useState("");
+
+    const handleTitleChange = (event) => {
+        let todoInput = document.querySelector("input");
+        let newTodoTitle = todoInput.value;
+        //let newTodoTitle = event.target.value;
+        setTodoTitle(newTodoTitle);
+    };
+
     const handleAddTodo = (event) => {
+        // let form = event.target;
         event.preventDefault();
-        console.log("Form Submitted!");
-        let form = event.target;
+        onAddTodo({id: Date.now(), title: todoTitle});
+        setTodoTitle("");
         let todoInput = document.querySelector("input");
```

Comment:
When using React, be sure that you are avoiding direct DOM manipulation. There are exceptions to this rule when using 3rd party libraries or certain event listeners but we won't run into any of those in this course.

In this case, the todoInput's value is already being managed by `todoTitle`. `useState` provides both a state value and a function to update that state value. When `setTodoTitle("some value") is called, `todoTitle` is updated. React will subsequently update the value inside the input element in the DOM. Direct updates will confuse the state inside of React and can cause bugs.

Since this PR is already closed, make this update inside of lesson 1.5.

Reply:
Thank you for the explanation. I have replaced the old Vanilla JavaScript code with code that has more to do with React, for the purposes of clearing the input field, once the user presses the “Add” button. I have replaced with a call to that “function to update that state value.” So, now, we are using the following code inside the declaration for the handleAddTodo function, instead of the old code, which directly manipulated the DOM:
`setTodoTitle("");`